File: backend/app/services/organization_service.py
from app.db.supabase import supabase


class OrganizationService:

    @staticmethod
    def create_organization(
        name: str,
        organization_type: str,
        registration_no: str,
        address: str | None = None,
        phone: str | None = None,
    ):
        data = {
            "name": name,
            "type": organization_type,
            "registration_no": registration_no,
            "address": address,
            "phone": phone,
        }

        response = (
            supabase
            .table("organizations")
            .insert(data)
            .execute()
        )

        return response.data

    @staticmethod
    def get_organization_by_id(organization_id: str):
        response = (
            supabase
            .table("organizations")
            .select("*")
            .eq("id", organization_id)
            .single()
            .execute()
        )

        return response.data

    @staticmethod
    def get_organization_by_registration_no(
        registration_no: str
    ):
        response = (
            supabase
            .table("organizations")
            .select("*")
            .eq(
                "registration_no",
                registration_no
            )
            .single()
            .execute()
        )

        return response.data

    @staticmethod
    def get_all_organizations():
        response = (
            supabase
            .table("organizations")
            .select("*")
            .execute()
        )

        return response.data
    @staticmethod
    def get_organizations_by_type(
        organization_type: str
    ):
        response = (
            supabase
            .table("organizations")
            .select("*")
            .eq("type", organization_type)
            .execute()
        )

        return response.data

File: frontend/src/services/organizationService.js
import axios from "axios";

const API = axios.create({
    baseURL: "http://127.0.0.1:8000",
});


// Get all organizations
export const getAllOrganizations = async () => {
    const response = await API.get("/organizations/");
    return response.data;
};


// Get organization by ID
export const getOrganization = async (id) => {
    const response = await API.get(`/organizations/${id}`);
    return response.data;
};


// Create organization
export const createOrganization = async (organizationData) => {
    const response = await API.post(
        "/organizations/",
        organizationData
    );

    return response.data;
};


// Get organizations by type
export const getOrganizationsByType = async (type) => {
    const response = await API.get(
        `/organizations/by-type?type=${type}`
    );

    return response.data;
};